IPs/RSKIP419.md

```diff
@@ -0,0 +1,233 @@
+The funding transaction hash signed being saved means that the proof Transaction can be created.
+
+**Important remark**: this method is public, meaning any user could call it.
+
+#### Proof transaction creation or validation process failure
+
+A new method, `processPowpegChangeValidation` will be called from the `updateCollections` method.
+
+The `processPowpegChangeValidation` will perform some actions if a proposedFederation exists depending on this two scenarios:
+- SVP period ended.
+Since the proposedFederation still exists, this implies the proof transaction was not received, and therefore that the validation was not successful
+    -  The Bridge will emit a new event,
+        
+        `COMMIT_FEDERATION_FAILED(blockNumber indexed uint)`
+    - The proposed Powpeg will be eliminated.
+    - The election will be allowed once again.
+
+- SVP period still ongoing        
+    - If the funding transaction signed hash is registered in the SVP_FUND_TX_HASH_SIGNED storage entry, the Bridge will:
+        - Create it as described [here](#proof-transaction)
+        - Add it to a new storage set, key `spendTxWaitingForSignatures`
+        - Set it in a new Storage entry:
+            ```
+            KEY: "SVP_SPEND_TX_HASH"
+            VALUE: hash of SVP proof tx without signatures
+            ```
+        - Remove the SVP_FUND_TX_HASH_SIGNED from the storage    
+
+##### Proof transaction
+
+The Proof transaction is a Bitcoin transaction where an UTXO from the proposed Powpeg is sent to the current Powpeg.
+This transaction will be as simple and small as possible to avoid spending much satoshis as fees.
+
+Proof transaction structure:
+```
+{
+    inputs: [
+        {
+            txid: SVP_FUND_TX_HASH_SIGNED,
+            output_index: SVP_FUNDING_TX_OUTPUT_INDEX
+        }
+    ],
+    outputs: [
+        {
+            script: CURRENT_POWPEG_OUTPUT_SCRIPT,
+            value: TOTAL_UTXO - FEES
+        }
+    ]
+}
+```
+
+#### Proof transaction confirmation
+
+The SVP proof transaction should go through a new "peg-out signing process", since it has to be signed for the proposed pegnatories which belong neither the active nor the retiring powpeg, but the proposed one.
+
+#### Proof transaction signing
+
+The proof transaction should be signed as any other peg-out, but by the proposed powpeg. For this to happen, a new Bridge method `addSpendTxSignature` should be created to perform the exact same actions as the `addSignature` but to accept signatures from public keys that don't belong to an active/retiring Powpeg (let's recall the proposed Powpeg is not active at this point).
+The `addSpendTxSignature` method should check that the public key belongs to the proposed Powpeg and that the hash of the transaction intended to be signed exists in the `SVP_SPEND_TX_HASH` storage entry.
+
+Once fully signed, the Bridge should remove the entry from the `spendTxWaitingForSignatures` and emit the release_btc event as usual.
```

Comment:
Not sure about emitting `release_btc` event here. It is not a release or pegout, it is the proposed federation returning the funds. If someone is tracking that event it will create confusion

Reply:
is true